### DISCUSSION TITLE
(Contribution) New API  member Must Exist Error on Build

### DISCUSSION THREAD

**Discussion by sps014** [dotnet/runtime#58353] [Q&A] [answered]
Is there any procedure that one must keep in mind while adding a new API?

I have added a Member in IList Interface and implemented this in relevant classes.
Now when I build the.net, I received the following error.
```
MembersMustExist : Member 'public void System.Collections.Generic.List<T>.RemoveAt(System.Index)' 
does not exist in the reference but it does exist in the implementation.
```
I added the above error in `runtime\src\libraries\System.Collections\src\MatchingRefApiCompatBaseline.txt` as was suggested in the previous full error.
Now I am receiving the following error
```
microsoft.dotnet.apicompat\6.0.0-beta.21425.3\build\Microsoft.DotNet.ApiCompat.targets(82,5): error :
 InterfacesShouldHaveSameMembers : Interface member 'public void System.Collections.IList.RemoveAt(System.Index)' is present 
in the implementation but not in the contract. 
[C:\Users\shive\Desktop\Projects\runtime\src\libraries\System.Runtime\src\System.Runtime.csproj]
C:\Users\shive\.nuget\packages\microsoft.dotnet.apicompat\6.0.0-beta.21425.3\build\Microsoft.DotNet.ApiCompat.targets(96,5): 
error : ApiCompat failed for 'C:\Users\shive\Desktop\Projects\runtime\artifacts\bin\System.Runtime\net6.0-Debug\System.Runtime.dll'
 [C:\Users\shive\Desktop\Projects\runtime\src\libraries\System.Runtime\src\System.Runtime.csproj]
```

**Comment by gfoidl** [comment, marked as answer]
See [Updating Ref source](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/updating-ref-source.md).

But keep in mind, that new APIs need to be approved first. Please check [API Review Process](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md) for more infos on that side.

**Reply by sps014** [reply]
Thanks
